orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -712,11 +712,24 @@ class TaskController {
     return aStartTime <=> bStartTime ?: b.id <=> a.id
   }
 
+  static Boolean shouldReplace(Map.Entry entry, Map variables) {
+    if (!variables.containsKey(entry.key)) {
+      // an original key should always be saved
+      return true
+    }
+    // a duplicate key with an empty value should
+    // not overwrite a previous key with a non empty value
+    def currentVal = variables[entry.key]
+    return !(!isNullOrEmpty(currentVal) && isNullOrEmpty(entry.value))
+  }
+
   private OrchestrationViewModel convert(Execution orchestration) {
     def variables = [:]
     for (stage in orchestration.stages) {
       for (entry in stage.context.entrySet()) {
-        variables[entry.key] = entry.value
+        if (shouldReplace(entry, variables)) {
```

Comment:
Since `Map` returns `null` for nonexistent keys, I feel this would be easier to read (and no need for `shouldReplace` function)
```suggestion
        if (isNullOrEmpty(variables[entry.key]) || !isNullOrEmpty(entry.value)) {
```

Reply:
👍 that is better

Reply:
Used this logic but kept it in a separate function for easy testing.